src/components/Menus/DropdownMenu.js

```diff
@@ -0,0 +1,207 @@
+    /** Requesting to open/close for controlled open component. */
+    onRequestChange: PropTypes.func,
+    /** Callback function fired when a MenuItem is selected */
+    onSelect: PropTypes.func,
+    /** Customize style of menu parent */
+    style: PropTypes.shape({}),
+    /** Props passed to Menu component */
+    menuProps: PropTypes.shape({})
+  }
+
+  static defaultProps = {
+    open: null
+  }
+
+  state = {
+    open: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open !== null && this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.open()
+      } else {
+        this.close()
+      }
+    }
+  }
+
+  _handleSelect = (option) => {
+    const { onSelect, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(false)
+    } else {
+      // if multi then do not close
+      this.close()
+    }
+
+    onSelect && onSelect(option)
+  }
+
+  _handleClick = (event) => {
+    event.preventDefault()
+    const { open, onClick, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(!open)
+      return
+    }
+
+    if (!this.state.open) {
+      this.open()
+    } else {
+      this.close()
+    }
+    onClick && onClick(event)
+  }
+
+  _handleMenuBlur = () => {
+    const { onRequestChange } = this.props
+    // Timeout ensures click event has correct state when attempting to close
+    setTimeout( () => {
+      if (this.controlledOpen()) {
+        onRequestChange && onRequestChange(false)
+      } else {
+        this.close()
+      }
+    }, 100)
+  }
+
+  _handleKeyDown = (event) => {
+    const { onRequestChange } = this.props
+    switch(event.key) {
+      case 'Enter':
+        if (!event.defaultPrevented && this.controlledOpen()) {
+          event.preventDefault()
+          onRequestChange && onRequestChange(true)
+        } else {
+          event.preventDefault()
+          if (this.state.open) {
+            this.close()
+          } else {
+            this.open()
+          }
+        }
+        break
+      case 'Escape':
+        event.preventDefault()
+
+        if (this.controlledOpen()) {
+          onRequestChange && onRequestChange(false)
+        } else {
+          this.state.open && this.close()
+        }
+        break
+    }
+  }
+
+  controlledOpen() {
+    return this.props.open !== null
+  }
+
+  open() {
+    const { onOpen } = this.props
+
+    this.setState({open: true}, () => {
+      this.menu.focus()
+      onOpen && onOpen()
+    })
+  }
+
+  close() {
+    const { onClose } = this.props
+    this.setState({open: false}, () => {
+      this.menu.blur()
+      onClose && onClose()
+    })
+  }
+
+  renderTriggerElement() {
+    const { triggerElement } = this.props
+    const { open } = this.state
+
+    if (triggerElement) {
+      return React.cloneElement(triggerElement, {
+        onClick: this._handleClick,
```

Comment:
This seems like unexpected behavior. I wouldn't assume that the `triggerElement` I passed in would lose it's click handler when passed into the component, can we just surround the `triggerElement` with a another div that we give the click handler too?